druid/src/widget/images.rs

```diff
@@ -0,0 +1,229 @@
+// Copyright 2020 The xi-editor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! An Image widget.
+//! Please consider using SVG and the SVG wideget as it scales much better.
+
+use std::error::Error;
+use std::marker::PhantomData;
+use std::str::FromStr;
+
+use image;
+
+use crate::{
+    Affine, BoxConstraints, Data, Env, Event, EventCtx, LayoutCtx, LifeCycle, LifeCycleCtx,
+    PaintCtx, Point, Rect, RenderContext, Size, UpdateCtx, Widget,
+};
+
+use crate::piet::{ImageFormat, InterpolationMode};
+
+#[derive(PartialEq)]
+pub enum FillStrat {
+    Contain,
+    Cover,
+    Fill,
+    FitHeight,
+    FitWidth,
+    None,
+    ScaleDown,
+}
+
+impl Default for FillStrat {
+    fn default() -> Self {
+        FillStrat::Fill
+    }
+}
+
+fn get_scale_offset(parent: Size, fit_box: Size, fit_type: &FillStrat) -> (Point, Point) {
```

Comment:
Instead of returning (`Point`, `Point`), I would just return an `Affine`? We also have a [`TranslateScale`](https://docs.rs/kurbo/0.5.8/kurbo/struct.TranslateScale.html) type in kurbo, but that only does uniform scaling, but it isn't clear that we need non-uniform scaling?

Reply:
Some of the fill strategies do, I think a Affine transform would have more degrees of freedom than needed which alway feels wrong to me.

Reply:
umm these two are turned in to a affine before being used so it would simplify the api but at the expense of the api having too many degrees of freedom.

A affine is also sub optimal from the memory requirements of the api but on a x86/aarch64 that massively premature optimisation to worry.

Reply:
I think the simpler/clearer API is perhaps a better tradeoff, but we're splitting hairs.

I also think `Vec2` is a more appropriate type than `Point` as written, but if you're happy I'm happy.